File: service/jmj-provider-api/jmj-provider-api-hall/src/main/java/com/wllfengshu/jmj/provider/api/hall/HallService.java
package com.wllfengshu.jmj.provider.api.hall;

import com.wllfengshu.jmj.provider.api.hall.model.givehall.GiveHallRequest;
import com.wllfengshu.jmj.provider.api.hall.model.givehall.GiveHallResponse;
import io.swagger.annotations.ApiOperation;
import org.springframework.cloud.openfeign.FeignClient;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;

/**
 * @author wangll
 * @date 2022-02-13 20:04
 */
@FeignClient(name = "jmj-provider-hall")
public interface HallService {

    @ApiOperation("获取大厅")
    @PostMapping(value = "/giveHall")
    GiveHallResponse giveHall(@RequestBody GiveHallRequest request);


}

File: client/assets/scripts/server/HallService.ts
import BaseService from "./BaseService";

export default class HallService extends BaseService {
     private static readonly serverName : string = "jmj-provider-hall";
     public static readonly instance : HallService = new HallService();

     // 获取大厅
     public giveHall(request : GiveHallRequest) : GiveHallResponse {
          return <GiveHallResponse> JSON.parse(super.callInterface(HallService.serverName, "giveHall", request));
     }

}

